docs/dev/kube-apiserver-health-check.md

```diff
@@ -0,0 +1,82 @@
+## Graceful Termination
+`kube-apiserver` in OpenShift is fronted by an external and an internal Load Balancer. This document serves as a
+guideline on how to properly configure the health check probe of the load balancers so that when a `kube-apiserver`
+instance restarts we can ensure:
+- The load balancers detect it and takes it out of service in time. No new request should be forwarded to the 
+  `kube-apiserver` instance when it has stopped listening.
+- Existing connections are not cut off hard, they are allowed to complete gracefully.
+
+## Load Balancer Health Check Probe
+`kube-apiserver` provides graceful termination support via the `/readyz` health check endpoint. When `/readyz` reports
+`ok` it indicates that the apiserver is ready to serve request(s).
+
+Now let's walk through the events (in chronological order) that unfold when a `kube-apiserver` instance restarts:
+* E1: `T+0s`: `kube-apiserver` receives a TERM signal.
+* E2: `T+0s`: `/readyz` starts reporting `failure` to signal to the load balancers that a shut down is in progress.
+  * The apiserver will continue to accept new request(s).
+  * The apiserver waits for certain amount of time (configurable by `shutdown-delay-duration`) before it stops accepting new request(s).
+* E3: `T+70s`: `kube-apiserver` (the http server) stops listening:
```

Comment:
Elsewhere in the doc you have:

> In future we will reduce `shutdown-delay-duration` to `30s`.

I'd rather make this portion of the doc robust to that sort of pivot by using `T+shudown-delay-duration` here.

Reply:
As far as the user/dev is concerned, they should treat `shutdown-delay-duration` to be `30s` for the purpose of designing health check probes. So I changed it to `T+30s`.